File: lab3/s1/ejb/src/main/java/ru/uvuv643/ejb/remote/dto/human/ListHumanBeingDto.java
package ru.uvuv643.ejb.remote.dto.human;

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;

import java.io.Serializable;
import java.util.List;

@JacksonXmlRootElement(localName = "ListHumanBeingDto")
public class ListHumanBeingDto implements Serializable {

        @JacksonXmlProperty(namespace = "objects")
        private List<HumanBeingDto> humanBeing;

        public ListHumanBeingDto() {
                this(null);
        }

        public ListHumanBeingDto(List<HumanBeingDto> humanBeing) {
                this.humanBeing = humanBeing;
        }

        public List<HumanBeingDto> getHumanBeing() {
                return humanBeing;
        }

        public void setHumanBeing(List<HumanBeingDto> humanBeing) {
                this.humanBeing = humanBeing;
        }
}

File: lab2/s2/src/main/kotlin/ru/uvuv643/soa/api/v1/dto/human/ListHumanBeingDto.kt
package ru.uvuv643.soa.api.v1.dto.human

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.xml.bind.annotation.XmlAccessType
import jakarta.xml.bind.annotation.XmlAccessorType
import jakarta.xml.bind.annotation.XmlElementWrapper
import jakarta.xml.bind.annotation.XmlRootElement

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
data class ListHumanBeingDto(

        @XmlElementWrapper(name = "pairs")
        val humanBeingPair : List<PairTeamHumanDto>,

) {
        constructor() : this(listOf())
}